src/server/server.ts

```diff
@@ -573,7 +573,7 @@ namespace ts.server {
         }
 
         private writeToEventSocket(body: any, eventName: string): void {
-            this.eventSocket.write(formatMessage(toEvent(body, eventName), this.logger, this.byteLength, this.host.newLine), "utf8");
+            this.eventSocket.write(formatMessage(toEvent(eventName, body), this.logger, this.byteLength, this.host.newLine), "utf8");

```

Comment:
type of `body` should be `object` on twoEvent, as well as on `writeToEventSocket`. that should flag invalid uses like these in the future.

Reply:
not `{}`, `object`..

Reply:
`{}` is the top-type.. a `string` is assignable to `{}` since it has no required properties.. 

`object` is the non-primitive object, i.e. something that is not `string`, `number`, `boolean`, or `symbol`

Reply:
Both callers have `body: T`, which seems not to be assignable to `object`.  Do I need to make an additional change to make that work?